.github/workflows/publish-snapshot.yml

```diff
@@ -5,6 +5,10 @@ on:
     branches:
       - develop
 
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
+  
```

Comment:
I'd also skip using these for these kinds of publishing jobs.

Maybe we should only do this for PR checks? It doesn't seem super useful for other workflows, and if a release/publish kind of workflow is cancelled while it's in the middle of execution (I don't know how cancellation works exactly), that could leave things in a weird state.

Reply:
It is canceling the previous job (The same name) on the current branch, but always run the last one. So we shouldn't leave things in a weird state